clients/cloudsourceapi/client.go

```diff
@@ -0,0 +1,131 @@
+package cloudsourceapi
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/estafette/estafette-ci-api/config"
+	"golang.org/x/oauth2"
+	sourcerepo "google.golang.org/api/sourcerepo/v1"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// Client is the interface for communicating with the Google Cloud Source Repository api
+type Client interface {
+	GetAccessToken(ctx context.Context) (accesstoken AccessToken, err error)
+	GetAuthenticatedRepositoryURL(ctx context.Context, accesstoken AccessToken, htmlURL string) (url string, err error)
+	GetEstafetteManifest(ctx context.Context, accesstoken AccessToken, notification PubSubNotification) (valid bool, manifest string, err error)
+	JobVarsFunc(ctx context.Context) func(ctx context.Context, repoSource, repoOwner, repoName string) (token string, url string, err error)
+}
+
+// NewClient creates an githubapi.Client to communicate with the Google Cloud Source Repository api
+func NewClient(config config.CloudSourceConfig, sourcerepoService *sourcerepo.Service, tokenSource oauth2.TokenSource) Client {
+
+	return &client{
+		service:     sourcerepoService,
+		config:      config,
+		tokenSource: tokenSource,
+	}
+}
+
+type client struct {
+	service     *sourcerepo.Service
+	config      config.CloudSourceConfig
+	tokenSource oauth2.TokenSource
+}
+
+// GetAccessToken returns an access token to access the Bitbucket api
+func (c *client) GetAccessToken(ctx context.Context) (accesstoken AccessToken, err error) {
+
+	token, err := c.tokenSource.Token()
+	if err != nil {
+		return
+	}
+
+	accesstoken = AccessToken{
+		AccessToken:  token.AccessToken,
+		Expiry:       token.Expiry,
+		RefreshToken: token.RefreshToken,
+		TokenType:    token.TokenType,
+	}
+
+	return
+}
+
+func (c *client) GetAuthenticatedRepositoryURL(ctx context.Context, accesstoken AccessToken, htmlURL string) (url string, err error) {
+
+	url = strings.Replace(htmlURL, "https://source.developers.google.com", fmt.Sprintf("https://x-token-auth:%v@source.developers.google.com", accesstoken.AccessToken), -1)
+
+	return
+}
+
+func (c *client) GetEstafetteManifest(ctx context.Context, accesstoken AccessToken, notification PubSubNotification) (exists bool, manifest string, err error) {
+
+	repoSource := notification.GetRepoSource()
+	repoOwner := notification.GetRepoOwner()
+	repoName := notification.GetRepoName()
+	repoRefName := ""
+	for _, refUpdate := range notification.RefUpdateEvent.RefUpdates {
+		repoRefName = refUpdate.RefName
+	}
+
+	// Tempdir to clone the repository
+	dir, err := ioutil.TempDir("", "sourcerepo-manifest")
+	if err != nil {
+		return
+	}
+
+	defer os.RemoveAll(dir) // clean up
+
+	// Clones the repository into the given dir, just as a normal git clone does
+	_, err = git.PlainClone(dir, false, &git.CloneOptions{
```

Comment:
Is this a shallow clone?

Reply:
It can be, I'll add the depth option. I think 10?

Reply:
In the `git-clone` extension the default is 50, see https://github.com/estafette/estafette-extension-git-clone/blob/master/main.go#L33